### ISSUE TITLE
Rotating Points Object

### ISSUE THREAD

**Issue by cbravo** [toxicFork/react-three-renderer#97]
I am not sure what I am doing wrong here. According to the documentation Points should be able to take a rotation parameter (https://github.com/toxicFork/react-three-renderer/wiki/points). But I don't see that reflected in the ThreeJS documentation here (http://threejs.org/docs/?q=points#Reference/Objects/Points). However I have seen a bunch of examples of rotating point clouds  (http://threejs.org/examples/?q=points#webgl_points_sprites) 

My render function looks something like this (sorry in advance for not having a code pen I am on a deadline and throwing a hailmary)...  and I am not seeing any errors. It just seems to not want to rotate. 

I can see (via console log) that my render function is running again and again and the values for scale and rotation are changing but visually I am not seeing any difference. What am I missing?

```
update(){
   let { rotation, scale } = this.state;
   rotation.y += 0.1;
   scale.x += 0.1;
   scale.y += 0.1;
   scale.z += 0.1;
   this.setState({
      rotation : rotation,
      scale: scale
    });
  }

render() {
    const {
      rotation,
      scale
    } = this.state;
    return (
      
      <points scale={scale} rotation={rotation}>
        <geometry vertices={this.pointCloudVertices}/>
        <pointsMaterial
          color={0xff0000}
        />
      </points>
    );
  }
```

**Comment by toxicFork** [comment]
Sounds like a bug, I will take a look

**Comment by cbravo** [comment]
Thank you for looking into this Toxic!!! Great library btw!

**Comment by toxicFork** [comment]
@cbravo can you try setting the state with new values instead, for example 
```js
this.setState({
    scale: new THREE.Vector3(newX, newY, newZ),
})
```

**Comment by cbravo** [comment]
I ended up mutating the rotation directly using refs which i am guessing is better performance... seems to be at least.

**Comment by toxicFork** [comment]
You sacrifice a bit of comfort for a bit of performance :) Also apologies for the late reply, I had been very busy even in weekends until now.

Looks like the case is safe to close, I will edit the title to redirect potential dupes

**Comment by toxicFork** [comment]
More info: react-three-renderer chooses not to update objects if the attributes remain the same, e.g. if you have:

```jsx
var myVec = new THREE.Vector3(1,2,3);
// ...

<mesh position={myVec} />

myVec.x = 5;

<mesh position={myVec} />

```

This may not update the position since the variable reference has not changed

The workaround mentioned above is:

```jsx
<mesh ref={myMeshRef} />

this.myMesh.position.x = 5;
```

This is considered against the react philosophy (it's similar to setting a div's attribute manually) but can result in performance benefits e.g. avoiding react lifecycles.

**Comment by superhighfives** [comment]
Hey @toxicFork. Awesome library!

I'm trying to create a floating mesh, but I've been unable to get the vertices to update:

```
import React from 'react'
import React3 from 'react-three-renderer'
import THREE from 'three'

class FloatingMesh extends React.Component {
  constructor (props, context) {
    super(props, context)

    this.animate = this.animate.bind(this)
    this.cameraPosition = new THREE.Vector3(0, 0, 100)
    this.frame = 0
  }

  animate () {
    this.frame = this.frame + 1
    let center = new THREE.Vector2(0, 0)
    let vertices = this.refs.plane.vertices.map((v) => {
      let dist = new THREE.Vector2(v.x, v.y).sub(center)
      let size = 5.0
      let magnitude = 10
      let z = Math.sin(dist.length() / (-size / 2) + ((this.frame) / 1000)) * magnitude
      return new THREE.Vector3(v.x, v.y, z)
    })
    this.refs.plane.vertices = vertices
    this.refs.plane.verticesNeedUpdate = true
  }

  render () {
    let width = window.innerWidth
    let height = window.innerHeight

    return (<React3 clearColor='#17293a' mainCamera='camera' width={width} height={height} onAnimate={this.animate}>
      <scene>
        <perspectiveCamera name='camera' fov={75} aspect={width / height} near={0.1} far={10000} position={this.cameraPosition} />
        <mesh>
          <planeGeometry ref='plane' width={200} height={100} widthSegments={20} heightSegments={20} />
          <meshBasicMaterial color={0x00ff00} wireframe wireframeLinewidth={0.1} />
        </mesh>
      </scene>
    </React3>)
  }
}

export default FloatingMesh
```

Tried it by setting `vertices={this.state.vertices}` on the planeGeometry component too (rather than setting it via `this.refs.plane.vertices`, but no change. Wondering if you had any thoughts on where I might be going wrong?

For reference, visually it stays looking like this:
![image](https://cloud.githubusercontent.com/assets/449385/19597990/f33b8262-978f-11e6-9da1-1b0680c0d69c.png)

Again, awesome library. 👍

**Comment by toxicFork** [comment]
Trying it here as well and cannot seem to update the vertices of the PlaneGeometry directly as well... I am wondering if it can be updated only once for some strange reason

**Comment by superhighfives** [comment]
I've set up a Codepen with vanilla Three over here if that helps at all: http://codepen.io/superhighfives/pen/qaQRGd?editors=0010

**Comment by toxicFork** [comment]
@superhighfives 

It looks like a bug or feature in threejs.

It does not let you assign new `.vertices` property into the geometry but you can edit in place.

For example this works:

```
  animate() {
    this.frame = this.frame + 1;
    const center = new THREE.Vector2(0, 0);

    this.refs.plane.vertices.forEach(v => {
      const dist = new THREE.Vector2(v.x, v.y).sub(center);
      const size = 5.0;
      const magnitude = 10;
      v.z = Math.sin(dist.length() / -size / 2 + (this.frame / 1000)) * magnitude;
    });

    this.refs.plane.verticesNeedUpdate = true;
  }
```

Looking at code in threejs:

```
	if ( object instanceof THREE.Mesh ) {

			var direct = geometry.__directGeometry;

			if ( direct === undefined ) {

				return this.fromGeometry( geometry );

			}

			direct.verticesNeedUpdate = geometry.verticesNeedUpdate;
			direct.normalsNeedUpdate = geometry.normalsNeedUpdate;
			direct.colorsNeedUpdate = geometry.colorsNeedUpdate;
			direct.uvsNeedUpdate = geometry.uvsNeedUpdate;
			direct.groupsNeedUpdate = geometry.groupsNeedUpdate;

			geometry.verticesNeedUpdate = false;
			geometry.normalsNeedUpdate = false;
			geometry.colorsNeedUpdate = false;
			geometry.uvsNeedUpdate = false;
			geometry.groupsNeedUpdate = false;

			geometry = direct;

		}

		if ( geometry.verticesNeedUpdate === true ) {

			var attribute = this.attributes.position;

			if ( attribute !== undefined ) {

				attribute.copyVector3sArray( geometry.vertices );
				attribute.needsUpdate = true;

			}

			geometry.verticesNeedUpdate = false;

		}
```

**Comment by superhighfives** [comment]
Amazing! Thank you so much. Do you have a tip jar somewhere?

Also, I noticed that the line widths seem different between the Codepen and the example - I tried adding `antialias` to the `React3` component, but there's still a difference between them. Wondering if there was something I was missing in smoothing or refining the `wireframeLinewidth`. Happy to open a new issue if that's easier.

Codepen:
![image](https://cloud.githubusercontent.com/assets/449385/19601712/58650c00-97a2-11e6-99b3-0f4ceac0ca9a.png)

Example:
![image](https://cloud.githubusercontent.com/assets/449385/19601724/64247fe4-97a2-11e6-8bb6-6423c91b8267.png)

And side by side:
![image](https://cloud.githubusercontent.com/assets/449385/19601763/85cb78d2-97a2-11e6-862e-618c26affa1a.png)

For both I have `wireframeLinewidth` set to 1.

**Comment by toxicFork** [comment]
Pay it forward :)

Yes please make a new issue, will be easier to investigate once I resolve
other bugs.
